### PR TITLE
DOC: Separate out examples - pandas.str.is methods

### LINKED ISSUE #59806: DOC: Separate Examples for String Methods (str.isalnum(), str.isalpha(), etc.) in docs
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.Series.str.isalnum.html

### Documentation problem

In the current documentation for pandas string methods `(e.g., str.isalnum(), str.isalpha(), str.isdigit(), etc.)`, the examples for all string methods are grouped together in a way that makes it difficult to interpret which example corresponds to which method. This can be confusing for users who are looking for examples for a specific method, and it reduces the clarity of the documentation.

### Suggested fix for documentation

I suggest updating the documentation to have distinct examples for each string method. Each method should have its own section with a description and example specific to that method, which would improve clarity and make it easier to navigate.

If this is deemed valid, I would be happy to work on resolving it. Could this issue be assigned to me?

### PR DESCRIPTION
**Pull request by saldanhad**
- [x] closes #59806
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by saldanhad** [comment]
Hi @rhshadrach, I have refactored the docstrings to separate the `Examples` and `See Also` sections for each `ismethod`, no code changes were made and tested the changes locally. However, I noticed that the unit tests are now failing, which seems to be related to an issue with the CI script. Could you kindly review the changes and provide your feedback when possible? Thank you.

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -3583,31 +3684,49 @@ def casefold(self):
...
     isalnum = _map_and_wrap(
-        "isalnum", docstring=_shared_docs["ismethods"] % _doc_args["isalnum"]
+        "isalnum",
+        docstring=_shared_docs["ismethods"] % _doc_args["isalnum"]
+        + _shared_docs["isalnum"],
     )
     isalpha = _map_and_wrap(
-        "isalpha", docstring=_shared_docs["ismethods"] % _doc_args["isalpha"]
+        "isalpha",
+        docstring=_shared_docs["ismethods"] % _doc_args["isdigit"]
```
Comment (rhshadrach) [inline pandas/core/strings/accessor.py:3693]: Should be isalpha instead of isdigit here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/strings/accessor.py (hunk 1 above)
Comment (saldanhad) [reply pandas/core/strings/accessor.py]: Thanks for your review, sorry missed this, fixed now.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @saldanhad!

**Pull request merged**